cmake/module/AddThreadsIfNeeded.cmake

```diff
@@ -1,21 +1,14 @@
-# Copyright (c) 2023 The Bitcoin Core developers
+# Copyright (c) 2023-present The Bitcoin Core developers
 # Distributed under the MIT software license, see the accompanying
-# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+# file COPYING or https://opensource.org/license/mit/.
 
-function(add_threads_if_needed)
-  # TODO: Not all targets, which will be added in the future,
-  #       require Threads. Therefore, a proper check will be
-  #       appropriate here.
+if(CMAKE_C_COMPILER_LOADED)
```

Comment:
re offline discussion. This is dead code, it can be dropped entirely.

Reply:
Thanks! The dead code has been dropped.